spec/query_spec.rb

```diff
@@ -994,4 +994,44 @@
       end
     end
   end
+
+  describe '#pagination_links?' do
+    subject { instance.pagination_links? }
+
+    context 'when pagination_links_on_demand' do
+      before { Graphiti.config.pagination_links_on_demand = true }
```

Comment:
Since the config acts as a singleton, we need to make sure to reset this. Otherwise it can leak into other unrelated tests. Something like this (you'll see elsewhere):

```ruby
around do |e|
  original = Graphiti.config.pagination_links_on_demand
  Graphiti.config.pagination_links_on_demand = true
  begin
    e.run
  ensure
    Graphiti.config.pagination_links_on_demand = original
  end
end
```

Reply:
Yeah, that makes sense. Updated.

Reply:
Is it okay now @richmolj? Sorry for this mistake, I saw that this was a singleton, but when I start to write the specs I completely forgot 😂

Reply:
No worries! Looks great! Thanks so much for the help ❤️